### PR TITLE
Add 'container' option to the popover service and directive

### PR DESCRIPTION
**Pull request by mstaicu**

### CODE REVIEW

File: dist/js/styleguide-components.js [hunk 1]
```diff
@@ -5465,6 +5465,9 @@ function getElementOptions(element) {
   if (element.dataset.image) {
     options.image = element.dataset.image;
   }
+  if (element.dataset.container) {
```
Comment (skywritergr) [inline dist/js/styleguide-components.js]: Should we use a `switch` statement here for all those ifs?

File: dist/js/styleguide-components.js [hunk 2]
```diff
@@ -5672,11 +5676,13 @@ function PopoverService() {
    * @return {String}           [Popover's new placement]
    */
   function checkPopoverPlacement(placement) {
+    var popoverContainerElement = getPopoverParentElement(elementPopoverOptions);
+    var coordinateComputeFunction = popoverContainerElement === BODY ? getBoundingOffset : getParentOffset;
```
Comment (skywritergr) [inline dist/js/styleguide-components.js]: I don't see the `getBoundingOffset` and `getParentOffset` in the scope of this function. Where are they coming from?

Comment (mstaicu) [reply dist/js/styleguide-components.js]: We're computing the coordinates relative to the viewport or relative to the first offset parent based on the option that we pass via the `container` property. The `getBoundingOffset` and `getParentOffset` are functions that do just that, and we're deciding which coordinate function we're using based on the value passed as the `container`. Those functions are not scoped in the same function, they are declared as part of the service.

File: dist/js/styleguide-components.js (hunk 1 above)
Comment (mstaicu) [reply dist/js/styleguide-components.js]: This is turning into a spaghetti code, I'm thinking of something right now.

File: src/help/pop-over/pop-over.directive.spec.js [hunk 3]
```diff
@@ -67,11 +67,12 @@ describe('Popover directive', function() {
...
         <a tw-pop-over \
           data-original-title='<span>Popover title</span>' \
-          data-content='Full description copy, explaining in more detail'\
-          data-image='http://www.transferwise.com/logo.png'\
+          data-content='Full description copy, explaining in more detail' \
+          data-container='body' \
+          data-image='http://www.transferwise.com/logo.png' \
           data-template='<div class=\"popover\"> \
                            <h3 class=\"popover-title\"></h3> \
-                           <img class=\"popover-image\" alt=\"Popover content\" /> \
```
Comment (StevePole) [inline src/help/pop-over/pop-over.directive.spec.js]: Why did you remove the closing tag?

Comment (mstaicu) [reply src/help/pop-over/pop-over.directive.spec.js]: The  tag represents what is known as a void element so called because it can't have any contents. Therefore there is no syntactic reason why it should need to be closed in HTML. If the element is one of the void elements, in our case ``, or if the element is a foreign element, then there may be a single "/" (U+002F) character. This character has no effect on void elements.

Comment (StevePole) [reply src/help/pop-over/pop-over.directive.spec.js]: Does htmllint agree?  If it passes that's fine.
